### PR TITLE
BUG: Fix dtype inconsistency when appending to empty DataFrame

### LINKED ISSUE #22621: Dtype inconsistency when appending to empty dataframe
#### Code Sample, a copy-pastable example if possible

```python
>>> pd.DataFrame().append({"a": 1}, ignore_index=True)
     a
0  1.0
```
#### Problem description

I would expect append not to change the dtype of the a column from `int` to `float`, and for the output to be identical to that of `pd.concat([pd.DataFrame(), pd.DataFrame([{"a": 1}])])`

#### Expected Output
```python
>>> pd.DataFrame().append({"a": 1}, ignore_index=True)
     a
0  1
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: 4612312783e88d0dcd9cadfaa07077b0b382c26a
python: 2.7.13.final.0
python-bits: 64
OS: Darwin
OS-release: 15.6.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: None.None

pandas: 0.24.0.dev0+547.g4612312
pytest: None
pip: 18.0
setuptools: 34.3.3
Cython: 0.25.2
numpy: 1.13.3
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 5.3.0
sphinx: None
patsy: None
dateutil: 2.6.0
pytz: 2017.3
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.9999999
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None
</details>

### PR DESCRIPTION
**Pull request by lowerthansound**
- [ ] closes #22621
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

*edit*: currently rewriting the DataFrame.append function

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_combine_concat.py [hunk 1]
```diff
@@ -198,6 +198,15 @@ def test_append_dtypes(self):
         expected = DataFrame({'bar': Series([Timestamp('20130101'), 1])})
         assert_frame_equal(result, expected)
 
+        # GH 22621
+        # when appending a Series to an empty DataFrame
+        # the result must maintain the Series dtype
+
+        df = DataFrame()
+        series = Series({'A': 1}, name='foobar', dtype='int64')
+        assert df.append(series).dtypes['A'] == 'int64'
```
[the remark this answers is not in the document]
Comment (araraonline) [reply pandas/tests/frame/test_combine_concat.py]: just did it

File: doc/source/whatsnew/v0.24.0.txt [hunk 2]
```diff
@@ -775,6 +775,7 @@ Reshaping
 - Bug in :meth:`DataFrame.replace` raises ``RecursionError`` when replacing empty lists (:issue:`22083`)
 - Bug in :meth:`Series.replace` and meth:`DataFrame.replace` when dict is used as the ``to_replace`` value and one key in the dict is is another key's value, the results were inconsistent between using integer key and using string key (:issue:`20656`)
 - Bug in :meth:`DataFrame.drop_duplicates` for empty ``DataFrame`` which incorrectly raises an error (:issue:`20516`)
+- Bug in :meth:`DataFrame.append` would produce an inconsistent dtype when appending to empty DataFrame (:issue:`22621`)
```
[the remark this answers is not in the document]
Comment (araraonline) [reply doc/source/whatsnew/v0.24.0.txt]: and this too!

**Comment by araraonline** [comment]
Got a question.

The problem ended up having a scope slightly bigger than what was mentioned in the Issue #22621

Now I'm working on a branch that fixes the dtype inconsistency when appending an array to a dframe, whether this dframe is empty or not.

Should I create a new PR?

**Comment by jreback** [comment]
@araraonline no you can just update this one

**Comment by araraonline** [comment]
I'm currently working on rewriting the DataFrame.append function. When it is ready, the issue related here will be resolved

**Pull request closed without merging**